sisyphus/graph.py

```diff
@@ -230,16 +230,20 @@ def output(self):
 
     def add_target(self, target):
         """
-        :param OutputTarget target:
+        :param OutputPath target:
         """
         self._targets.append(target)
 
         # check if output path is already used
         try:
-            path = target._output_path
-            if path in self.used_output_path:
-                logging.warning('Output path is used more than once: %s' % path)
-            self.used_output_path.add(path)
+            path = str(target._output_path)
+            if path in self.used_output_sis_path.keys() and self.used_output_sis_path[path] != target._sis_path:
```

Comment:
'if path in self.used_output_sis_path.keys()' should only be  'if path in self.used_output_sis_path'